openpype/hosts/fusion/plugins/publish/validate_saver_resolution.py

```diff
@@ -0,0 +1,95 @@
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin,
+)
+
+from openpype.hosts.fusion.api.action import SelectInvalidAction
+from openpype.hosts.fusion.api import (
+    get_current_comp,
+    comp_lock_and_undo_chunk,
+)
+
+
+class ValidateSaverResolution(
+    pyblish.api.InstancePlugin, OptionalPyblishPluginMixin
+):
+    """Validate that the saver input resolution matches the projects"""
+
+    order = pyblish.api.ValidatorOrder
+    label = "Validate Saver Resolution"
+    families = ["render"]
+    hosts = ["fusion"]
+    optional = True
+    actions = [SelectInvalidAction]
+
+    @classmethod
+    def get_invalid(cls, instance, wrong_resolution=None):
+        if wrong_resolution is None:
+            wrong_resolution = []
+
+        saver = instance[0]
+        firstFrame = saver.GetAttrs("TOOLNT_Region_Start")[1]
+        comp = get_current_comp()
+
+        # If the current saver hasn't bin rendered its input resolution
+        # hasn't bin saved. To combat this, add an expression in
+        # the comments field to read the resolution
+
+        # False undo removes the undo-stack from the undo list
+        with comp_lock_and_undo_chunk(comp, "Read resolution", False):
```

Comment:
I'm having a hard time understanding what is going on inside this `with` , probably cause i am not familiar with fusions API, but what confuses me is how in the same `saver["Comments"]` we get and set expressions.

Reply:
If the nodes isn't yet rendered (eg you just opened the comp) an expression is needed to "touch" the node-tree to get the calculated resolution.

`saver["Comments"].SetExpression("xxxx")` sets the expression to the comments controller. To read the result that the expression generates you need to do it at a frame (as the expressions result can change on different frames normally), thus `saver["Comments"][firstFrame]`

It's a bit of a hacky way but it's the only way I found with my extended research on how to get the resolution.

Reply:
Thanks for explaining :D